<html>
<head>
     <meta charset="UTF-8">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
    <script src="./js/main.js"></script>
    <script src="./js/header_data.js"></script>
    <script src="http://d3js.org/d3.v3.min.js" type="text/javascript"></script>
    <script src="./js/VCF2PopTree.js"></script>
    <script src="./js/google_analytics.js"></script>

    <link rel="stylesheet" href="./css/main.css">
    <title>SNPVersity 2.0</title>
</head>

<body>
    <div class="body-container">
       <div class="header">
           Fusarium SNPVersity: a web-based tool for visualizing <i>Fusarium</i> diversity
       </div>

       <div class="tabs">
           <div>
               <button class="tab" id="Soptions" onclick="showContent('input');highlightTab('Soptions')">Select options</button>
               <button class="tab" id="Voutput" onclick="showContent('output');highlightTab('Voutput')">Table view</button>
               <button class="tab" id="Toutput" onclick="showContent('outputTree');highlightTab('Toutput')">Tree view</button>
           </div>

           <button class="tab" id="download" onclick="showContent('download');highlightTab('download')">Downloads</button> <!-- Help tab on far right -->
           <button class="tab" id="help" onclick="showContent('help');highlightTab('help')">Help</button> <!-- Help tab on far right -->
       </div>

       <div id="mainContainer">
           <div id="pageContentsXX">
               <div id="contentContainer" class="content">
                    <h2><font color="green">Fusarium VCF builder and viewer</font></h2>
                    <p>Create a VCF file based on a given region of the <i>Fusarium</i> genome and a subset of the diverse strains.</p>

                    <div id="subheaderContainer">
                        <p><h3><font color="white">Select genomic interval</font></h3></p>
                    </div>

                    <p>To retrieve the SNPs and INDELS within a genomic interval, select a chromosome, then enter the genome start and stop positions of your interval.</p>
                    <form action="#" method="post">
                        <table width="2800px" class="input_table">

                          <tr>
                            <td width="20%"><b><font color="green">Chromosome:</font></b></td>
                            <td width="55%">
                              <select name='chr' id='chrInput'>
                                  <option value="chr1" selected>Chromosome 1</option>
                                  <option value="chr2">Chromosome 2</option>
                                  <option value="chr3">Chromosome 3</option>
                                  <option value="chr4">Chromosome 4</option>
                              </select>
                            </td>
                            <td width="25%"> </td>
                          </tr>
                          <tr>
                            <td><b><font color="green">Genome Start Position (bp):</font></b>
                            <td>
                                <input type="number"  style="width: 100px;" min="0" name="start" id="startInput" value="2520531"/>
                            </td>
                          </tr>
                          <tr>
                            <td><b><font color="green">Genome End Position (bp):</font></b></td>
                            <td>
                            <input type="number" style="width: 100px;" min="0" name="stop" id="endInput" value="2524130"/>
                            </td>
                            <td> </td>
                          </tr>
                          <tr>
                            <td><b><font color="green">Loci per page:</font></b></td>
                            <td>
                              <select id="pageInput" name='perpage'>
                                <option value="50">50</option>
                                <option value="100" selected>100</option>
                                <option value="250">250</option>
                                <option value="500">500</option>
                              </select>
                            </td>
                            <td> </td>
                          </tr>
                        </table>

                        <br>
                        Optional: Enter a gene model and press the "Load coordinates" button and the gene model coordinates will be auto-populated above.<br><br>

                        <table width="1500px" class="input_table">
                            <tr>
                                <td style="vertical-align: top; padding-top: 5px" width="20%"><b><font color="green">Gene Model ID:</font></b><br></td>
                                <td width="55%"> <input type="text" name="geneModelId" id="geneModelId"> (+/- bp) <input type="number"  style="width: 100px;" min="0" name="start" id="flankInput" value="0"/>
                                    <input type="button" value="Load coordinates" onclick="loadGeneModelData()"> (<a href="javascript:loadExample()">See Example</a>)
                                    <br>
                                    <span id="error"></span>
                                </td>
                                <td width="25%"> </td>
                             </tr>
                        </table>
                        <br>
                        <div id="subheaderContainer">
                            <p><h3><font color="white">Choose a dataset</font></h3></p>
                        </div>
                        <br>

                        Use the "Select" column to choose a dataset. Once you've made your selection, a list of available accessions will appear in the "Select the accessions to include" section.

                        <br>

                        <table class="dataset-table"> <!-- Added class "dataset-table" -->
                            <thead>
                                <tr>
                                    <th>Select</th>
                                    <th>Dataset</th>
                                    <th>Reference Genome</th>
                                    <th>Accessions</th>
                                    <th>Variant Sites</th>
                                    <th>Key Filters</th>
                                    <th>Heterozygous Sites</th>
                                    <th>INDELS</th>
                                    <th>Imputation</th>
                                    <th>Paper</th>
                                    <th>Notes</th>
                                </tr>
                            </thead>
                            <tbody>
                                <!-- Dataset 1 -->
                                <tr>
                                    <td class="radio-cell">
                                        <input type="radio" name="ds" id='dataset' value="mgdb2024_hq" checked>
                                    </td>
                                    <td><i>F. graminearum</i> 2024 (High Quality)</td>
                                    <td><i>Fusarium graminearum</i> PH-1 (NRRL 31084)</td>
                                    <td>513</td>
                                    <td>3.1 million</td>
                                    <td>Mapping quality (≥40), Genotypic coverage (≥ 50%), LD (max R² > 0.5)</td>
                                    <td><span class="crossmark">&#10008;</span></td>
                                    <td><span class="checkmark">&#10004;</span></td>
                                    <td><span class="crossmark">&#10008;</span></td>
                                    <td>
                                        In preperation.
                                    </td>
                                    <td>See the Help and Download tabs for more information about the dataset.</td>
                                </tr>

                                <!-- Dataset 2 -->
                                <tr>
                                    <td class="radio-cell">
                                        <input type="radio" name="ds" id='dataset' value="mgdb2024_hc">
                                    </td>
                                    <td><i>F. graminearum</i> 2024 (High Coverage)</td>
                                    <td><i>Fusarium graminearum</i> PH-1 (NRRL 31084)</td>
                                    <td>513</td>
                                    <td>3.5 million</td>
                                    <td>Mapping quality (≥ 40), Genotypic coverage (≥ 50%)</td>
                                    <td><span class="crossmark">&#10008;</span></td>
                                    <td><span class="checkmark">&#10004;</span></td>
                                    <td><span class="crossmark">&#10008;</span></td>
                                    <td>
                                        In preperation.
                                    </td>
                                    <td>See the Help and Download tabs for more information about the dataset.</td>
                                </tr>

                            </tbody>
                        </table>

                        <br>
                        <div id="subheaderContainer">
                            <p><h3><font color="white">Select the accessions to include</font></h3></p>
                        </div>

                        <p class="upload-instructions">
                          Upload a file of <i>Fusarium</i> species:
                        </p>

                        <div class="input_container">
                          <input type="file" id="fileUpload" name="fileUpload">
                          <button class="button_up" type="button" onclick="document.getElementById('fileUpload').value = '';">Clear</button>
                        </div>

                          <div style="margin-top: 20px;">
                            <p style="margin-bottom: 10px;">
                              The accession should listed in a single-column seperated by a tab or newline. Use the "SNPversity ID" from these lists of all <i>Fusarium</i> species:
                              <a href="fg_accessions.xlsx" download="fg_accessions.xlsx">XLSX</a>.
                            </p>
                          </div>

                        --OR--

                        <div class="maize2024_container" style="display: block;" >
                        <br>
                        <b><font color="green" style="font-size: 10pt;">Select the acccessions from the following projects:</font></b>

                        <br><br>
                        <button type="button" id="selectRandom" onclick="toggleCheckboxesAll(0,'.genotypes')">Unselect all</button>
                        <button type="button" id="selectRandom" onclick="toggleCheckboxesAll(0.05,'.genotypes')">Select Random 5%</button>
                        <button type="button" id="selectRandom" onclick="toggleCheckboxesAll(0.10,'.genotypes')">Select Random 10%</button>
                        <button type="button" id="selectRandom" onclick="toggleCheckboxesAll(0.25,'.genotypes')">Select Random 25%</button>
                        <button type="button" id="selectRandom" onclick="toggleCheckboxesAll(0.5,'.genotypes')">Select Random 50%</button>
                        <button type="button" id="selectRandom" onclick="toggleCheckboxesAll(0.75,'.genotypes')">Select Random 75%</button>
                        <button type="button" id="selectRandom" onclick="toggleCheckboxesAll(1.0,'.genotypes')">Select All</button>
                        <br><br>

                                <div class="border1">
                                    <button type="button" class="accordion"><b>F. graminearum, 513 strains - Location: Various</b></button>
                                    <div class="panel">
                                        USDA-ARS: 513 total strains

                                    <table>
                                        <tr>
                                            <td colspan="5">
                                                <button type="button" id="selectRandom" onclick="toggleCheckboxes(this, 0,'.genotypes')">Unselect all</button>
                                                <button type="button" id="selectRandom" onclick="toggleCheckboxes(this, 0.25,'.genotypes')">Select Random 25%</button>
                                                <button type="button" id="selectRandom" onclick="toggleCheckboxes(this, 0.5,'.genotypes')">Select Random 50%</button>
                                                <button type="button" id="selectRandom" onclick="toggleCheckboxes(this, 1.0,'.genotypes')">Select All</button>
                                            </td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="06163mAP" />06163mAP</td><td><input class="genotypes" type="checkbox" value="31619mAP" />31619mAP</td><td><input class="genotypes" type="checkbox" value="31620mAP" />31620mAP</td><td><input class="genotypes" type="checkbox" value="34302mAP" />34302mAP</td><td><input class="genotypes" type="checkbox" value="34494mAP" />34494mAP</td><td><input class="genotypes" type="checkbox" value="34540mAP" />34540mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="34583mAP" />34583mAP</td><td><input class="genotypes" type="checkbox" value="34584mAP" />34584mAP</td><td><input class="genotypes" type="checkbox" value="34585mAP" />34585mAP</td><td><input class="genotypes" type="checkbox" value="34586mAP" />34586mAP</td><td><input class="genotypes" type="checkbox" value="36905mAP" />36905mAP</td><td><input class="genotypes" type="checkbox" value="36970trimmAP" />36970trimmAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="37401mAP" />37401mAP</td><td><input class="genotypes" type="checkbox" value="37408mAP" />37408mAP</td><td><input class="genotypes" type="checkbox" value="37425mAP" />37425mAP</td><td><input class="genotypes" type="checkbox" value="37450mAP" />37450mAP</td><td><input class="genotypes" type="checkbox" value="37509mAP" />37509mAP</td><td><input class="genotypes" type="checkbox" value="37525mAP" />37525mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="38165mAP" />38165mAP</td><td><input class="genotypes" type="checkbox" value="38369mAP" />38369mAP</td><td><input class="genotypes" type="checkbox" value="38380mAP" />38380mAP</td><td><input class="genotypes" type="checkbox" value="38381mAP" />38381mAP</td><td><input class="genotypes" type="checkbox" value="38383mAP" />38383mAP</td><td><input class="genotypes" type="checkbox" value="38393mAP" />38393mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="38395mAP" />38395mAP</td><td><input class="genotypes" type="checkbox" value="38405mAP" />38405mAP</td><td><input class="genotypes" type="checkbox" value="39238mAP" />39238mAP</td><td><input class="genotypes" type="checkbox" value="39247mAP" />39247mAP</td><td><input class="genotypes" type="checkbox" value="39263mAP" />39263mAP</td><td><input class="genotypes" type="checkbox" value="39270mAP" />39270mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="39274mAP" />39274mAP</td><td><input class="genotypes" type="checkbox" value="39278mAP" />39278mAP</td><td><input class="genotypes" type="checkbox" value="39290mAP" />39290mAP</td><td><input class="genotypes" type="checkbox" value="39297mAP" />39297mAP</td><td><input class="genotypes" type="checkbox" value="39300mAP" />39300mAP</td><td><input class="genotypes" type="checkbox" value="39310mAP" />39310mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="39339mAP" />39339mAP</td><td><input class="genotypes" type="checkbox" value="39340mAP" />39340mAP</td><td><input class="genotypes" type="checkbox" value="39341mAP" />39341mAP</td><td><input class="genotypes" type="checkbox" value="39343mAP" />39343mAP</td><td><input class="genotypes" type="checkbox" value="43161mAP" />43161mAP</td><td><input class="genotypes" type="checkbox" value="43884mAP" />43884mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="44070mAP" />44070mAP</td><td><input class="genotypes" type="checkbox" value="44078mAP" />44078mAP</td><td><input class="genotypes" type="checkbox" value="44211mAP" />44211mAP</td><td><input class="genotypes" type="checkbox" value="45156mAP" />45156mAP</td><td><input class="genotypes" type="checkbox" value="45373mAP" />45373mAP</td><td><input class="genotypes" type="checkbox" value="45596mAP" />45596mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="45606mAP" />45606mAP</td><td><input class="genotypes" type="checkbox" value="45607mAP" />45607mAP</td><td><input class="genotypes" type="checkbox" value="45611mAP" />45611mAP</td><td><input class="genotypes" type="checkbox" value="45612mAP" />45612mAP</td><td><input class="genotypes" type="checkbox" value="45620mAP" />45620mAP</td><td><input class="genotypes" type="checkbox" value="45626mAP" />45626mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="45628mAP" />45628mAP</td><td><input class="genotypes" type="checkbox" value="45632mAP" />45632mAP</td><td><input class="genotypes" type="checkbox" value="45633mAP" />45633mAP</td><td><input class="genotypes" type="checkbox" value="45634mAP" />45634mAP</td><td><input class="genotypes" type="checkbox" value="45635mAP" />45635mAP</td><td><input class="genotypes" type="checkbox" value="45636mAP" />45636mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="45637mAP" />45637mAP</td><td><input class="genotypes" type="checkbox" value="45647mAP" />45647mAP</td><td><input class="genotypes" type="checkbox" value="45649mAP" />45649mAP</td><td><input class="genotypes" type="checkbox" value="45654mAP" />45654mAP</td><td><input class="genotypes" type="checkbox" value="45655mAP" />45655mAP</td><td><input class="genotypes" type="checkbox" value="45656mAP" />45656mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="45657mAP" />45657mAP</td><td><input class="genotypes" type="checkbox" value="45658mAP" />45658mAP</td><td><input class="genotypes" type="checkbox" value="45664mAP" />45664mAP</td><td><input class="genotypes" type="checkbox" value="45666mAP" />45666mAP</td><td><input class="genotypes" type="checkbox" value="45668mAP" />45668mAP</td><td><input class="genotypes" type="checkbox" value="45669mAP" />45669mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="45670mAP" />45670mAP</td><td><input class="genotypes" type="checkbox" value="45675mAP" />45675mAP</td><td><input class="genotypes" type="checkbox" value="45677mAP" />45677mAP</td><td><input class="genotypes" type="checkbox" value="45689mAP" />45689mAP</td><td><input class="genotypes" type="checkbox" value="45701mAP" />45701mAP</td><td><input class="genotypes" type="checkbox" value="45702mAP" />45702mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="45703mAP" />45703mAP</td><td><input class="genotypes" type="checkbox" value="45705mAP" />45705mAP</td><td><input class="genotypes" type="checkbox" value="45708mAP" />45708mAP</td><td><input class="genotypes" type="checkbox" value="45712mAP" />45712mAP</td><td><input class="genotypes" type="checkbox" value="45716mAP" />45716mAP</td><td><input class="genotypes" type="checkbox" value="45735mAP" />45735mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="45737mAP" />45737mAP</td><td><input class="genotypes" type="checkbox" value="45740mAP" />45740mAP</td><td><input class="genotypes" type="checkbox" value="45741mAP" />45741mAP</td><td><input class="genotypes" type="checkbox" value="45744mAP" />45744mAP</td><td><input class="genotypes" type="checkbox" value="45745mAP" />45745mAP</td><td><input class="genotypes" type="checkbox" value="45746mAP" />45746mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="45747mAP" />45747mAP</td><td><input class="genotypes" type="checkbox" value="45780mAP" />45780mAP</td><td><input class="genotypes" type="checkbox" value="45787mAP" />45787mAP</td><td><input class="genotypes" type="checkbox" value="45815mAP" />45815mAP</td><td><input class="genotypes" type="checkbox" value="45827mAP" />45827mAP</td><td><input class="genotypes" type="checkbox" value="45834mAP" />45834mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="45835mAP" />45835mAP</td><td><input class="genotypes" type="checkbox" value="45836mAP" />45836mAP</td><td><input class="genotypes" type="checkbox" value="45838mAP" />45838mAP</td><td><input class="genotypes" type="checkbox" value="45839mAP" />45839mAP</td><td><input class="genotypes" type="checkbox" value="46240mAP" />46240mAP</td><td><input class="genotypes" type="checkbox" value="46250mAP" />46250mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="46259mAP" />46259mAP</td><td><input class="genotypes" type="checkbox" value="46267mAP" />46267mAP</td><td><input class="genotypes" type="checkbox" value="46273mAP" />46273mAP</td><td><input class="genotypes" type="checkbox" value="46275mAP" />46275mAP</td><td><input class="genotypes" type="checkbox" value="46284mAP" />46284mAP</td><td><input class="genotypes" type="checkbox" value="46287mAP" />46287mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="46289mAP" />46289mAP</td><td><input class="genotypes" type="checkbox" value="46290mAP" />46290mAP</td><td><input class="genotypes" type="checkbox" value="46294mAP" />46294mAP</td><td><input class="genotypes" type="checkbox" value="46297mAP" />46297mAP</td><td><input class="genotypes" type="checkbox" value="46305mAP" />46305mAP</td><td><input class="genotypes" type="checkbox" value="46317mAP" />46317mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="46329mAP" />46329mAP</td><td><input class="genotypes" type="checkbox" value="46336mAP" />46336mAP</td><td><input class="genotypes" type="checkbox" value="46354mAP" />46354mAP</td><td><input class="genotypes" type="checkbox" value="46363mAP" />46363mAP</td><td><input class="genotypes" type="checkbox" value="46370mAP" />46370mAP</td><td><input class="genotypes" type="checkbox" value="46371mAP" />46371mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="46380mAP" />46380mAP</td><td><input class="genotypes" type="checkbox" value="46386mAP" />46386mAP</td><td><input class="genotypes" type="checkbox" value="46422mAP" />46422mAP</td><td><input class="genotypes" type="checkbox" value="46434mAP" />46434mAP</td><td><input class="genotypes" type="checkbox" value="46522mAP" />46522mAP</td><td><input class="genotypes" type="checkbox" value="46525mAP" />46525mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="46532mAP" />46532mAP</td><td><input class="genotypes" type="checkbox" value="46533mAP" />46533mAP</td><td><input class="genotypes" type="checkbox" value="46539mAP" />46539mAP</td><td><input class="genotypes" type="checkbox" value="46544mAP" />46544mAP</td><td><input class="genotypes" type="checkbox" value="46547mAP" />46547mAP</td><td><input class="genotypes" type="checkbox" value="46548mAP" />46548mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="46549mAP" />46549mAP</td><td><input class="genotypes" type="checkbox" value="46558mAP" />46558mAP</td><td><input class="genotypes" type="checkbox" value="46559mAP" />46559mAP</td><td><input class="genotypes" type="checkbox" value="46560mAP" />46560mAP</td><td><input class="genotypes" type="checkbox" value="46564mAP" />46564mAP</td><td><input class="genotypes" type="checkbox" value="46566mAP" />46566mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="46874mAP" />46874mAP</td><td><input class="genotypes" type="checkbox" value="46875mAP" />46875mAP</td><td><input class="genotypes" type="checkbox" value="46898mAP" />46898mAP</td><td><input class="genotypes" type="checkbox" value="46899mAP" />46899mAP</td><td><input class="genotypes" type="checkbox" value="46914mAP" />46914mAP</td><td><input class="genotypes" type="checkbox" value="46915mAP" />46915mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="46934mAP" />46934mAP</td><td><input class="genotypes" type="checkbox" value="46935mAP" />46935mAP</td><td><input class="genotypes" type="checkbox" value="46959mAP" />46959mAP</td><td><input class="genotypes" type="checkbox" value="46960mAP" />46960mAP</td><td><input class="genotypes" type="checkbox" value="46986mAP" />46986mAP</td><td><input class="genotypes" type="checkbox" value="46987mAP" />46987mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="47526mAP" />47526mAP</td><td><input class="genotypes" type="checkbox" value="47527mAP" />47527mAP</td><td><input class="genotypes" type="checkbox" value="47554mAP" />47554mAP</td><td><input class="genotypes" type="checkbox" value="47555mAP" />47555mAP</td><td><input class="genotypes" type="checkbox" value="47556mAP" />47556mAP</td><td><input class="genotypes" type="checkbox" value="47571mAP" />47571mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="47588mAP" />47588mAP</td><td><input class="genotypes" type="checkbox" value="47605mAP" />47605mAP</td><td><input class="genotypes" type="checkbox" value="47659mAP" />47659mAP</td><td><input class="genotypes" type="checkbox" value="52005mAP" />52005mAP</td><td><input class="genotypes" type="checkbox" value="52008mAP" />52008mAP</td><td><input class="genotypes" type="checkbox" value="52129mAP" />52129mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="52195mAP" />52195mAP</td><td><input class="genotypes" type="checkbox" value="52429mAP" />52429mAP</td><td><input class="genotypes" type="checkbox" value="52512mAP" />52512mAP</td><td><input class="genotypes" type="checkbox" value="52840mAP" />52840mAP</td><td><input class="genotypes" type="checkbox" value="52847mAP" />52847mAP</td><td><input class="genotypes" type="checkbox" value="52850mAP" />52850mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="52955mAP" />52955mAP</td><td><input class="genotypes" type="checkbox" value="53046mAP" />53046mAP</td><td><input class="genotypes" type="checkbox" value="53173mAP" />53173mAP</td><td><input class="genotypes" type="checkbox" value="53184mAP" />53184mAP</td><td><input class="genotypes" type="checkbox" value="54196mAP" />54196mAP</td><td><input class="genotypes" type="checkbox" value="54197mAP" />54197mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="62913mAP" />62913mAP</td><td><input class="genotypes" type="checkbox" value="62929mAP" />62929mAP</td><td><input class="genotypes" type="checkbox" value="66030mAP" />66030mAP</td><td><input class="genotypes" type="checkbox" value="66031mAP" />66031mAP</td><td><input class="genotypes" type="checkbox" value="66037mAP" />66037mAP</td><td><input class="genotypes" type="checkbox" value="66039mAP" />66039mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="66040mAP" />66040mAP</td><td><input class="genotypes" type="checkbox" value="66041mAP" />66041mAP</td><td><input class="genotypes" type="checkbox" value="66042mAP" />66042mAP</td><td><input class="genotypes" type="checkbox" value="66043mAP" />66043mAP</td><td><input class="genotypes" type="checkbox" value="66045mAP" />66045mAP</td><td><input class="genotypes" type="checkbox" value="66047mAP" />66047mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="66049mAP" />66049mAP</td><td><input class="genotypes" type="checkbox" value="ETH-001-barcode10mAP" />ETH-001-barcode10mAP</td><td><input class="genotypes" type="checkbox" value="ETH-001-barcode11mAP" />ETH-001-barcode11mAP</td><td><input class="genotypes" type="checkbox" value="ETH-001-barcode12mAP" />ETH-001-barcode12mAP</td><td><input class="genotypes" type="checkbox" value="ETH-001-barcode7mAP" />ETH-001-barcode7mAP</td><td><input class="genotypes" type="checkbox" value="ETH-001-barcode8mAP" />ETH-001-barcode8mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="ETH-001-barcode9mAP" />ETH-001-barcode9mAP</td><td><input class="genotypes" type="checkbox" value="F100mAP" />F100mAP</td><td><input class="genotypes" type="checkbox" value="F10mAP" />F10mAP</td><td><input class="genotypes" type="checkbox" value="F116mAP" />F116mAP</td><td><input class="genotypes" type="checkbox" value="F124mAP" />F124mAP</td><td><input class="genotypes" type="checkbox" value="F130mAP" />F130mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="F131mAP" />F131mAP</td><td><input class="genotypes" type="checkbox" value="F132mAP" />F132mAP</td><td><input class="genotypes" type="checkbox" value="F13mAP" />F13mAP</td><td><input class="genotypes" type="checkbox" value="F16mAP" />F16mAP</td><td><input class="genotypes" type="checkbox" value="F272mAP" />F272mAP</td><td><input class="genotypes" type="checkbox" value="F27mAP" />F27mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="F322mAP" />F322mAP</td><td><input class="genotypes" type="checkbox" value="F325mAP" />F325mAP</td><td><input class="genotypes" type="checkbox" value="F326mAP" />F326mAP</td><td><input class="genotypes" type="checkbox" value="F327mAP" />F327mAP</td><td><input class="genotypes" type="checkbox" value="F328mAP" />F328mAP</td><td><input class="genotypes" type="checkbox" value="F329mAP" />F329mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="F330mAP" />F330mAP</td><td><input class="genotypes" type="checkbox" value="F331mAP" />F331mAP</td><td><input class="genotypes" type="checkbox" value="F332mAP" />F332mAP</td><td><input class="genotypes" type="checkbox" value="F333mAP" />F333mAP</td><td><input class="genotypes" type="checkbox" value="F334mAP" />F334mAP</td><td><input class="genotypes" type="checkbox" value="F335mAP" />F335mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="F336mAP" />F336mAP</td><td><input class="genotypes" type="checkbox" value="F337mAP" />F337mAP</td><td><input class="genotypes" type="checkbox" value="F338mAP" />F338mAP</td><td><input class="genotypes" type="checkbox" value="F339mAP" />F339mAP</td><td><input class="genotypes" type="checkbox" value="F340mAP" />F340mAP</td><td><input class="genotypes" type="checkbox" value="F341mAP" />F341mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="F342mAP" />F342mAP</td><td><input class="genotypes" type="checkbox" value="F343mAP" />F343mAP</td><td><input class="genotypes" type="checkbox" value="F344mAP" />F344mAP</td><td><input class="genotypes" type="checkbox" value="F347mAP" />F347mAP</td><td><input class="genotypes" type="checkbox" value="F348mAP" />F348mAP</td><td><input class="genotypes" type="checkbox" value="F351mAP" />F351mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="F353mAP" />F353mAP</td><td><input class="genotypes" type="checkbox" value="F354mAP" />F354mAP</td><td><input class="genotypes" type="checkbox" value="F356mAP" />F356mAP</td><td><input class="genotypes" type="checkbox" value="F366mAP" />F366mAP</td><td><input class="genotypes" type="checkbox" value="F367mAP" />F367mAP</td><td><input class="genotypes" type="checkbox" value="F368mAP" />F368mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="F387mAP" />F387mAP</td><td><input class="genotypes" type="checkbox" value="F398mAP" />F398mAP</td><td><input class="genotypes" type="checkbox" value="F399mAP" />F399mAP</td><td><input class="genotypes" type="checkbox" value="F501mAP" />F501mAP</td><td><input class="genotypes" type="checkbox" value="F522mAP" />F522mAP</td><td><input class="genotypes" type="checkbox" value="F535mAP" />F535mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="F537mAP" />F537mAP</td><td><input class="genotypes" type="checkbox" value="F55mAP" />F55mAP</td><td><input class="genotypes" type="checkbox" value="F562mAP" />F562mAP</td><td><input class="genotypes" type="checkbox" value="F60mAP" />F60mAP</td><td><input class="genotypes" type="checkbox" value="F68mAP" />F68mAP</td><td><input class="genotypes" type="checkbox" value="F6mAP" />F6mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="F96mAP" />F96mAP</td><td><input class="genotypes" type="checkbox" value="FG4mAP" />FG4mAP</td><td><input class="genotypes" type="checkbox" value="FHK10mAP" />FHK10mAP</td><td><input class="genotypes" type="checkbox" value="FHK11mAP" />FHK11mAP</td><td><input class="genotypes" type="checkbox" value="FHK12mAP" />FHK12mAP</td><td><input class="genotypes" type="checkbox" value="FHK13mAP" />FHK13mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK14mAP" />FHK14mAP</td><td><input class="genotypes" type="checkbox" value="FHK15mAP" />FHK15mAP</td><td><input class="genotypes" type="checkbox" value="FHK16mAP" />FHK16mAP</td><td><input class="genotypes" type="checkbox" value="FHK17mAP" />FHK17mAP</td><td><input class="genotypes" type="checkbox" value="FHK18mAP" />FHK18mAP</td><td><input class="genotypes" type="checkbox" value="FHK19mAP" />FHK19mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK1mAP" />FHK1mAP</td><td><input class="genotypes" type="checkbox" value="FHK20mAP" />FHK20mAP</td><td><input class="genotypes" type="checkbox" value="FHK21mAP" />FHK21mAP</td><td><input class="genotypes" type="checkbox" value="FHK22mAP" />FHK22mAP</td><td><input class="genotypes" type="checkbox" value="FHK23mAP" />FHK23mAP</td><td><input class="genotypes" type="checkbox" value="FHK24mAP" />FHK24mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK25mAP" />FHK25mAP</td><td><input class="genotypes" type="checkbox" value="FHK26mAP" />FHK26mAP</td><td><input class="genotypes" type="checkbox" value="FHK27mAP" />FHK27mAP</td><td><input class="genotypes" type="checkbox" value="FHK28mAP" />FHK28mAP</td><td><input class="genotypes" type="checkbox" value="FHK29mAP" />FHK29mAP</td><td><input class="genotypes" type="checkbox" value="FHK2mAP" />FHK2mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK30mAP" />FHK30mAP</td><td><input class="genotypes" type="checkbox" value="FHK31mAP" />FHK31mAP</td><td><input class="genotypes" type="checkbox" value="FHK32mAP" />FHK32mAP</td><td><input class="genotypes" type="checkbox" value="FHK33mAP" />FHK33mAP</td><td><input class="genotypes" type="checkbox" value="FHK34mAP" />FHK34mAP</td><td><input class="genotypes" type="checkbox" value="FHK35mAP" />FHK35mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK36mAP" />FHK36mAP</td><td><input class="genotypes" type="checkbox" value="FHK37mAP" />FHK37mAP</td><td><input class="genotypes" type="checkbox" value="FHK38mAP" />FHK38mAP</td><td><input class="genotypes" type="checkbox" value="FHK39mAP" />FHK39mAP</td><td><input class="genotypes" type="checkbox" value="FHK3mAP" />FHK3mAP</td><td><input class="genotypes" type="checkbox" value="FHK40mAP" />FHK40mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK41mAP" />FHK41mAP</td><td><input class="genotypes" type="checkbox" value="FHK42mAP" />FHK42mAP</td><td><input class="genotypes" type="checkbox" value="FHK43mAP" />FHK43mAP</td><td><input class="genotypes" type="checkbox" value="FHK44mAP" />FHK44mAP</td><td><input class="genotypes" type="checkbox" value="FHK45mAP" />FHK45mAP</td><td><input class="genotypes" type="checkbox" value="FHK46mAP" />FHK46mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK47mAP" />FHK47mAP</td><td><input class="genotypes" type="checkbox" value="FHK48mAP" />FHK48mAP</td><td><input class="genotypes" type="checkbox" value="FHK49mAP" />FHK49mAP</td><td><input class="genotypes" type="checkbox" value="FHK4mAP" />FHK4mAP</td><td><input class="genotypes" type="checkbox" value="FHK50mAP" />FHK50mAP</td><td><input class="genotypes" type="checkbox" value="FHK51mAP" />FHK51mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK52mAP" />FHK52mAP</td><td><input class="genotypes" type="checkbox" value="FHK53mAP" />FHK53mAP</td><td><input class="genotypes" type="checkbox" value="FHK54mAP" />FHK54mAP</td><td><input class="genotypes" type="checkbox" value="FHK55mAP" />FHK55mAP</td><td><input class="genotypes" type="checkbox" value="FHK56mAP" />FHK56mAP</td><td><input class="genotypes" type="checkbox" value="FHK57mAP" />FHK57mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK58mAP" />FHK58mAP</td><td><input class="genotypes" type="checkbox" value="FHK59mAP" />FHK59mAP</td><td><input class="genotypes" type="checkbox" value="FHK5mAP" />FHK5mAP</td><td><input class="genotypes" type="checkbox" value="FHK60mAP" />FHK60mAP</td><td><input class="genotypes" type="checkbox" value="FHK61mAP" />FHK61mAP</td><td><input class="genotypes" type="checkbox" value="FHK62mAP" />FHK62mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK63mAP" />FHK63mAP</td><td><input class="genotypes" type="checkbox" value="FHK64mAP" />FHK64mAP</td><td><input class="genotypes" type="checkbox" value="FHK65mAP" />FHK65mAP</td><td><input class="genotypes" type="checkbox" value="FHK66mAP" />FHK66mAP</td><td><input class="genotypes" type="checkbox" value="FHK67mAP" />FHK67mAP</td><td><input class="genotypes" type="checkbox" value="FHK68mAP" />FHK68mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK69mAP" />FHK69mAP</td><td><input class="genotypes" type="checkbox" value="FHK6mAP" />FHK6mAP</td><td><input class="genotypes" type="checkbox" value="FHK70mAP" />FHK70mAP</td><td><input class="genotypes" type="checkbox" value="FHK71mAP" />FHK71mAP</td><td><input class="genotypes" type="checkbox" value="FHK72mAP" />FHK72mAP</td><td><input class="genotypes" type="checkbox" value="FHK73mAP" />FHK73mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK74mAP" />FHK74mAP</td><td><input class="genotypes" type="checkbox" value="FHK75mAP" />FHK75mAP</td><td><input class="genotypes" type="checkbox" value="FHK76mAP" />FHK76mAP</td><td><input class="genotypes" type="checkbox" value="FHK77mAP" />FHK77mAP</td><td><input class="genotypes" type="checkbox" value="FHK78mAP" />FHK78mAP</td><td><input class="genotypes" type="checkbox" value="FHK79mAP" />FHK79mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK7mAP" />FHK7mAP</td><td><input class="genotypes" type="checkbox" value="FHK80mAP" />FHK80mAP</td><td><input class="genotypes" type="checkbox" value="FHK81mAP" />FHK81mAP</td><td><input class="genotypes" type="checkbox" value="FHK82mAP" />FHK82mAP</td><td><input class="genotypes" type="checkbox" value="FHK83mAP" />FHK83mAP</td><td><input class="genotypes" type="checkbox" value="FHK84mAP" />FHK84mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK85mAP" />FHK85mAP</td><td><input class="genotypes" type="checkbox" value="FHK86mAP" />FHK86mAP</td><td><input class="genotypes" type="checkbox" value="FHK87mAP" />FHK87mAP</td><td><input class="genotypes" type="checkbox" value="FHK89mAP" />FHK89mAP</td><td><input class="genotypes" type="checkbox" value="FHK8mAP" />FHK8mAP</td><td><input class="genotypes" type="checkbox" value="FHK90mAP" />FHK90mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK92mAP" />FHK92mAP</td><td><input class="genotypes" type="checkbox" value="FHK93mAP" />FHK93mAP</td><td><input class="genotypes" type="checkbox" value="FHK94mAP" />FHK94mAP</td><td><input class="genotypes" type="checkbox" value="FHK95mAP" />FHK95mAP</td><td><input class="genotypes" type="checkbox" value="FHK96mAP" />FHK96mAP</td><td><input class="genotypes" type="checkbox" value="FHK97mAP" />FHK97mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FHK9mAP" />FHK9mAP</td><td><input class="genotypes" type="checkbox" value="FRA12mAP" />FRA12mAP</td><td><input class="genotypes" type="checkbox" value="FRA180mAP" />FRA180mAP</td><td><input class="genotypes" type="checkbox" value="FRA200mAP" />FRA200mAP</td><td><input class="genotypes" type="checkbox" value="FRA21mAP" />FRA21mAP</td><td><input class="genotypes" type="checkbox" value="FRA305mAP" />FRA305mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="FRA31mAP" />FRA31mAP</td><td><input class="genotypes" type="checkbox" value="FRA65mAP" />FRA65mAP</td><td><input class="genotypes" type="checkbox" value="Fg1mAP" />Fg1mAP</td><td><input class="genotypes" type="checkbox" value="MTD41mAP" />MTD41mAP</td><td><input class="genotypes" type="checkbox" value="MTD66mAP" />MTD66mAP</td><td><input class="genotypes" type="checkbox" value="MTD69mAP" />MTD69mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="MTD71mAP" />MTD71mAP</td><td><input class="genotypes" type="checkbox" value="MTD72mAP" />MTD72mAP</td><td><input class="genotypes" type="checkbox" value="MTD80mAP" />MTD80mAP</td><td><input class="genotypes" type="checkbox" value="MTD83mAP" />MTD83mAP</td><td><input class="genotypes" type="checkbox" value="MTD84mAP" />MTD84mAP</td><td><input class="genotypes" type="checkbox" value="PH-1ERR11748137mAP" />PH-1ERR11748137mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN02585717mAP" />SAMN02585717mAP</td><td><input class="genotypes" type="checkbox" value="SAMN02981337mAP" />SAMN02981337mAP</td><td><input class="genotypes" type="checkbox" value="SAMN14604992mAP" />SAMN14604992mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774461mAP" />SAMN16774461mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774462mAP" />SAMN16774462mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774464mAP" />SAMN16774464mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774466mAP" />SAMN16774466mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774467mAP" />SAMN16774467mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774468mAP" />SAMN16774468mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774469mAP" />SAMN16774469mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774470mAP" />SAMN16774470mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774471mAP" />SAMN16774471mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774472mAP" />SAMN16774472mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774473mAP" />SAMN16774473mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774474mAP" />SAMN16774474mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774475mAP" />SAMN16774475mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774476mAP" />SAMN16774476mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774477mAP" />SAMN16774477mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774478mAP" />SAMN16774478mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774479mAP" />SAMN16774479mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774480mAP" />SAMN16774480mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774481mAP" />SAMN16774481mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774482mAP" />SAMN16774482mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774483mAP" />SAMN16774483mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774484mAP" />SAMN16774484mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774485mAP" />SAMN16774485mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774486mAP" />SAMN16774486mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774487mAP" />SAMN16774487mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774488mAP" />SAMN16774488mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774489mAP" />SAMN16774489mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774490mAP" />SAMN16774490mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774491mAP" />SAMN16774491mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774492mAP" />SAMN16774492mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774493mAP" />SAMN16774493mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774494mAP" />SAMN16774494mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774495mAP" />SAMN16774495mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774496mAP" />SAMN16774496mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774497mAP" />SAMN16774497mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774498mAP" />SAMN16774498mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774499mAP" />SAMN16774499mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774500mAP" />SAMN16774500mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774501mAP" />SAMN16774501mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774502mAP" />SAMN16774502mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774503mAP" />SAMN16774503mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774504mAP" />SAMN16774504mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774505mAP" />SAMN16774505mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774506mAP" />SAMN16774506mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774507mAP" />SAMN16774507mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774508mAP" />SAMN16774508mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774509mAP" />SAMN16774509mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774510mAP" />SAMN16774510mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774511mAP" />SAMN16774511mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774512mAP" />SAMN16774512mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774513mAP" />SAMN16774513mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774514mAP" />SAMN16774514mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774515mAP" />SAMN16774515mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774516mAP" />SAMN16774516mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774517mAP" />SAMN16774517mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774518mAP" />SAMN16774518mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774519mAP" />SAMN16774519mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774520mAP" />SAMN16774520mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774521mAP" />SAMN16774521mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774523mAP" />SAMN16774523mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774524mAP" />SAMN16774524mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774525mAP" />SAMN16774525mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774526mAP" />SAMN16774526mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774527mAP" />SAMN16774527mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774528mAP" />SAMN16774528mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774529mAP" />SAMN16774529mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774530mAP" />SAMN16774530mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774531mAP" />SAMN16774531mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774532mAP" />SAMN16774532mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774533mAP" />SAMN16774533mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774534mAP" />SAMN16774534mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774535mAP" />SAMN16774535mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774536mAP" />SAMN16774536mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774537mAP" />SAMN16774537mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774538mAP" />SAMN16774538mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774539mAP" />SAMN16774539mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774540mAP" />SAMN16774540mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774541mAP" />SAMN16774541mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774542mAP" />SAMN16774542mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774543mAP" />SAMN16774543mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774544mAP" />SAMN16774544mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774545mAP" />SAMN16774545mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774546mAP" />SAMN16774546mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774547mAP" />SAMN16774547mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774548mAP" />SAMN16774548mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774549mAP" />SAMN16774549mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774550mAP" />SAMN16774550mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="SAMN16774551mAP" />SAMN16774551mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774553mAP" />SAMN16774553mAP</td><td><input class="genotypes" type="checkbox" value="SAMN16774554mAP" />SAMN16774554mAP</td><td><input class="genotypes" type="checkbox" value="WG15AG37_S51mAP" />WG15AG37_S51mAP</td><td><input class="genotypes" type="checkbox" value="WG15BrC10_S49mAP" />WG15BrC10_S49mAP</td><td><input class="genotypes" type="checkbox" value="WG15BrK07_S48mAP" />WG15BrK07_S48mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="WG15BrK10_S47mAP" />WG15BrK10_S47mAP</td><td><input class="genotypes" type="checkbox" value="WG15CC02_S46mAP" />WG15CC02_S46mAP</td><td><input class="genotypes" type="checkbox" value="WG15EC187_S45mAP" />WG15EC187_S45mAP</td><td><input class="genotypes" type="checkbox" value="WG15EC232_S44mAP" />WG15EC232_S44mAP</td><td><input class="genotypes" type="checkbox" value="WG15EC233_S43mAP" />WG15EC233_S43mAP</td><td><input class="genotypes" type="checkbox" value="WG15EC234_S42mAP" />WG15EC234_S42mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="WG15EC236_S41mAP" />WG15EC236_S41mAP</td><td><input class="genotypes" type="checkbox" value="WG15EC239_S40mAP" />WG15EC239_S40mAP</td><td><input class="genotypes" type="checkbox" value="WG15EC256_S39mAP" />WG15EC256_S39mAP</td><td><input class="genotypes" type="checkbox" value="WG15EV127_S38mAP" />WG15EV127_S38mAP</td><td><input class="genotypes" type="checkbox" value="WG15EV156_S37mAP" />WG15EV156_S37mAP</td><td><input class="genotypes" type="checkbox" value="WG15EV202_S36mAP" />WG15EV202_S36mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="WG15EV224_S35mAP" />WG15EV224_S35mAP</td><td><input class="genotypes" type="checkbox" value="WG15EV246_S1mAP" />WG15EV246_S1mAP</td><td><input class="genotypes" type="checkbox" value="WG15EV247_S34mAP" />WG15EV247_S34mAP</td><td><input class="genotypes" type="checkbox" value="WG15Evill16_S33mAP" />WG15Evill16_S33mAP</td><td><input class="genotypes" type="checkbox" value="WG15Evill18_S32mAP" />WG15Evill18_S32mAP</td><td><input class="genotypes" type="checkbox" value="WG15Evill25_S31mAP" />WG15Evill25_S31mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="WG15Evill29_S30mAP" />WG15Evill29_S30mAP</td><td><input class="genotypes" type="checkbox" value="WG15Evill34_S29mAP" />WG15Evill34_S29mAP</td><td><input class="genotypes" type="checkbox" value="WG15Evill44_S28mAP" />WG15Evill44_S28mAP</td><td><input class="genotypes" type="checkbox" value="WG15GG06_S27mAP" />WG15GG06_S27mAP</td><td><input class="genotypes" type="checkbox" value="WG15GG11_S2mAP" />WG15GG11_S2mAP</td><td><input class="genotypes" type="checkbox" value="WG15GG17_S3mAP" />WG15GG17_S3mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="WG15GG23_S4mAP" />WG15GG23_S4mAP</td><td><input class="genotypes" type="checkbox" value="WG15GG39_S5mAP" />WG15GG39_S5mAP</td><td><input class="genotypes" type="checkbox" value="WG15GG42_S6mAP" />WG15GG42_S6mAP</td><td><input class="genotypes" type="checkbox" value="WG15HJ11_S7mAP" />WG15HJ11_S7mAP</td><td><input class="genotypes" type="checkbox" value="WG15HJ15_S8mAP" />WG15HJ15_S8mAP</td><td><input class="genotypes" type="checkbox" value="WG15HJ17_S9mAP" />WG15HJ17_S9mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="WG15HJ23_S10mAP" />WG15HJ23_S10mAP</td><td><input class="genotypes" type="checkbox" value="WG15HJ24_S11mAP" />WG15HJ24_S11mAP</td><td><input class="genotypes" type="checkbox" value="WG15HJ33_S12mAP" />WG15HJ33_S12mAP</td><td><input class="genotypes" type="checkbox" value="WG15HS21_S26mAP" />WG15HS21_S26mAP</td><td><input class="genotypes" type="checkbox" value="WG15HS32_S13mAP" />WG15HS32_S13mAP</td><td><input class="genotypes" type="checkbox" value="WG15LO06_S15mAP" />WG15LO06_S15mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="WG15PA02_S16mAP" />WG15PA02_S16mAP</td><td><input class="genotypes" type="checkbox" value="WG15PA17_S17mAP" />WG15PA17_S17mAP</td><td><input class="genotypes" type="checkbox" value="WG15PA21_S55mAP" />WG15PA21_S55mAP</td><td><input class="genotypes" type="checkbox" value="WG15PV109_S18mAP" />WG15PV109_S18mAP</td><td><input class="genotypes" type="checkbox" value="WG15PV132_S19mAP" />WG15PV132_S19mAP</td><td><input class="genotypes" type="checkbox" value="WG15PV59_S20mAP" />WG15PV59_S20mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="WG15PV63_S21mAP" />WG15PV63_S21mAP</td><td><input class="genotypes" type="checkbox" value="WG15PV65_S54mAP" />WG15PV65_S54mAP</td><td><input class="genotypes" type="checkbox" value="WG15PV66_S53mAP" />WG15PV66_S53mAP</td><td><input class="genotypes" type="checkbox" value="WG15PV72_S22mAP" />WG15PV72_S22mAP</td><td><input class="genotypes" type="checkbox" value="WG15PV73_S23mAP" />WG15PV73_S23mAP</td><td><input class="genotypes" type="checkbox" value="WG15PV75_S24mAP" />WG15PV75_S24mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="WG15PV82_S25mAP" />WG15PV82_S25mAP</td><td><input class="genotypes" type="checkbox" value="WG15SS01_S52mAP" />WG15SS01_S52mAP</td><td><input class="genotypes" type="checkbox" value="f688mAP" />f688mAP</td><td><input class="genotypes" type="checkbox" value="fra132mAP" />fra132mAP</td><td><input class="genotypes" type="checkbox" value="fra161mAP" />fra161mAP</td><td><input class="genotypes" type="checkbox" value="fra181mAP" />fra181mAP</td>
                                        </tr>
                                        <tr>
                                            <td><input class="genotypes" type="checkbox" value="fra201mAP" />fra201mAP</td><td><input class="genotypes" type="checkbox" value="fra267mAP" />fra267mAP</td><td><input class="genotypes" type="checkbox" value="fra336mAP" />fra336mAP</td>
                                        </tr>
                                    </table>

                                    <br>
                                </div>
                            </div>
    </div>
    <br><br>
    <input id="thevcfbuttons" type="submit" style="font-face: 'Comic Sans MS'; font-size: larger; color: black border: 3pt ridge lightgrey" value=" Create and view the VCF file! " name = "gene" >
    </input>
</div>
</div>
</div>
</form>



            <div id="loadingContainer" class="load_content" style="display: none">
            </div>
            <div id="outputContainer" class="content" style="display: none">
                Use the 'Select options' tab to query the WGS database.
            </div>



            <div id="treeContainer" class="content" style="display: none">

                <h2>Phylogenetic tree view powered by <a target="_blank" href="https://github.com/sansubs/vcf2pop">VCF2PopTree</a></h2>
                <div style="display: none">
              	<p>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp<font color="Green"><b>Read VCF file</b></font><br><br>

              	&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp  <input type="radio" name="colors3" id="pink">Compressed VCF file (vcf.gz)

              	&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp  <input type="radio" name="colors3" id="cyan" checked>VCF file (.vcf)<br>

              	<p>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbspFilter Single Nucleotide Variants (SNVs) by:</p>

                &nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp Minimum quality score <input type="text" size="5" onfocus="myFunc3()" id="Qtext" style="background-color:white;border:1px solid blue">&nbsp Minimum coverage depth <input type="text" size="5" onfocus="myFunc3()" id="Ctext" style="background-color:white;border:1px solid blue"><br><br>

              	&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp <input type="file" id="files" onclick="this.value=null;" size="52">&nbsp&nbsp&nbsp&nbsp&nbsp  <button onclick=goBack()>CLEAR</button><br><hr font color="blue">
                </div>
                <div style="display: none">
              	<p>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbspDistance&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp

            	&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp  <input type="radio" name="dis1" id="GD" checked>Genetic distance&nbsp&nbsp&nbsp <input type="radio" name="dis1" id="DD">Drift distance<br>

            	<p>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbspMissing Data&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp

            	&nbsp&nbsp  <input type="radio" name="site" id="PW" checked>Use SNVs for each pair of genomes&nbsp&nbsp&nbsp <input type="radio" name="site" id="CS">Use SNVs present in all genomes<br>

            	<p>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbspModel&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp

            	<input type="radio" name="EMD" id="evol1" onchange="myFunc1()" checked>Number of differences

            	&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp

            	<input type="radio" name="EMD" id="evol2" onchange="myFunc1()">P-distance

            	&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp

            	<input type="radio" name="EMD" id="evol3" onchange="myFunc1()">Jukes-Cantor distance<br>

            	<p id="par1">&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbspSize of the whole genome or chromosome (base pairs) <input type="text" id="myText" style="background-color:white;border:1px solid blue">
                </div>
                <table class="tree_table">
                    <tr>
                        <td>
                            Construct Tree
                        </td>
                        <td>
                            <input type="radio" name="colors1" id="black" checked>UPGMA tree <input type="radio" name="colors1" id="brown">Neighbour-Joining tree (Unrooted)<br>
                        </td>
                    </tr>
                    <tr>
                        <td>
                            Drawing options
                        </td>
                        <td>
                            <input type="radio" name="colors2" id="red" checked>Rectangular tree  <input type="radio" name="colors2" id="blue">Radial tree<br>
                        </td>
                    </tr>
                    <tr>
                        <td>
                            Output format
                        </td>
                        <td>
                            <input type="radio" name="colors2" id="green">Newick tree <input type="radio" name="colors2" id="yellow">Pair-wise diversity (MEGA) <input type="radio" name="colors2" id="orange">PHYLIP<br>
                        </td>
                    </tr>
                </table>

                <div style="display: none">
            	<p>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp<font color="Green"><b>Select genomes</b></font>

              	&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp  <input type="radio" name="ls1" onchange="myFunc2()" id="All" checked>All

            	&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp  <input type="radio" name="ls1" onchange="myFunc2()" id="Select">Selected

            	<p id="par2">&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp<font color="green" size="3">Enter the <b>exact names</b> of at least 4 genomes <b>- one per row</b></font><br>

            	&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp  <TEXTAREA id="PopList" rows="4" cols="50" style="background-color:white;border:1px solid blue" onfocus="checkRadio()" onfocusout=uncheckRadio(document.getElementById("PopList").value)></TEXTAREA><br></p>
                </div>
                <br>

                <div id="tree_message" style="display: block">
                No VCF file loaded. Use the 'Select options' tab to query the WGS database.
               </div>

                <div id="tree_button" style="display: none">
                    <button id="thetreebuttons" onclick="outputs(document.getElementById('black').checked, document.getElementById('brown').checked, document.getElementById('red').checked, document.getElementById('blue').checked, document.getElementById('green').checked, document.getElementById('yellow').checked, document.getElementById('orange').checked, document.getElementById('All').checked, document.getElementById('Select').checked, document.getElementById('PopList').value, document.getElementById('myText').value, document.getElementById('evol1').checked, document.getElementById('evol2').checked, document.getElementById('evol3').checked, document.getElementById('CS').checked, document.getElementById('PW').checked)">
                     Draw tree
                    </button>

                </div>
                <div id="tree_block" style="display: none">

                </div>
                   <div id="tree_output" style="display: none">
                       <div id='phylogram'></div>
                   </div>

            </div>

            <div id="downloadContainer" class="content" style="display: none">
                <h1>Fusarium SNPversity Data Downloads</h1>
                Release 1.0.01 (28 November 2024)

                <h2>Overview</h2>
                Below you will find detailed information about the datasets available for download, along with guidelines and best practices. For more information on how the datasets were generated see the 'Help' section.

                <h2>How to Download</h2>
                <ul class="info-list">
                    <li><b>Filtered datasets (Direct download)</b>: For the filtered datasets, click the link to the <a target="_blank" href="https://ars-usda.app.box.com/v/maizegdb-public/folder/255390517505">Federal Box account</a> and identify the files to download. Check the boxes next to the files and click the download icon near the top of the page. Alternatively, click on the filename listed in one of the tables below, which is linked to the Box download page for that file. </li>
                    <li><b>Unfiltered datasets (Request Access)</b>: Send an email to <a href=mailto:john.portwood@usda.gov>John Portwood</a> with the subject "WGS file access request". Once your request is received, you will receive a link and temporary password via email. This link will be active for 48 hours.</li>
                </ul>

                <h2>Download Instructions</h2>
                <ul class="info-list">
                    <li><b>Prepare Adequate Storage</b>: Ensure you have sufficient storage space on your system for the download.</li>
                    <li><b>Use a Reliable Internet Connection</b>: A wired connection is recommended for faster and more stable downloads.</li>
                    <li><b>Consider Using a Download Manager</b>: For large files, a download manager can help manage the download process, allowing for pause, resume, and recovery of the download.</li>
                    <li><b>Verify Data Integrity</b>: After download, use checksums (provided in the tables below) to verify the integrity of the downloaded files.</li>
                </ul>

                <h2>F. graminearum 2024 High Coverage Datasets (VCFs)</h2>

                <table width="80%" style="border-collapse: collapse; font-family: Arial, sans-serif; width: 90%">
                    <tr>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Dataset Name</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Description</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Size</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">*Download time (@100Mbps / @10MBps)</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">MD5 Checksum</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;"><a target="_blank" href="#">chr1_high_coverage.vcf.gz</a> </td>
                        <td style="padding: 10px;">High Coverage variants and metadata identified in Chromsome 1.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr >
                        <td style="padding: 10px;"><a target="_blank" href="#">chr2_high_coverage.vcf.gz</a> </td>
                        <td style="padding: 10px;">High Coverage variants and metadata identified in Chromsome 2.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;"><a target="_blank" href="#">chr3_high_coverage.vcf.gz</a> </td>
                        <td style="padding: 10px;">High Coverage variants and metadata identified in Chromsome 3.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr >
                        <td style="padding: 10px;"><a target="_blank" href="#">chr4_high_coverage.vcf.gz</a> </td>
                        <td style="padding: 10px;">High Coverage variants and metadata identified in Chromsome 4.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>

                </table>
                *Download times are estimated and may vary.

                <h2>F. graminearum 2024 High Coverage Datasets (H5)</h2>

                <table width="80%" style="border-collapse: collapse; font-family: Arial, sans-serif; width: 90%">
                    <tr>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Dataset Name</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Description</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Size</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">*Download time (@100Mbps / @10MBps)</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">MD5 Checksum</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;"><a target="_blank" href="#">fusarium2024_chr1_HC.h5.gz</a> </td>
                        <td style="padding: 10px;">High Coverage variants and metadata identified in Chromsome 1.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr >
                        <td style="padding: 10px;"><a target="_blank" href="#">fusarium2024_chr2_HC.h5.gz</a> </td>
                        <td style="padding: 10px;">High Coverage variants and metadata identified in Chromsome 2.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;"><a target="_blank" href="#">fusarium2024_chr3_HC.h5.gz</a> </td>
                        <td style="padding: 10px;">High Coverage variants and metadata identified in Chromsome 3.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr >
                        <td style="padding: 10px;"><a target="_blank" href="#">fusarium2024_chr4_HC.h5.gz</a> </td>
                        <td style="padding: 10px;">High Coverage variants and metadata identified in Chromsome 4.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>

                </table>
                *Download times are estimated and may vary.

                <h2>F. graminearum 2024 High Quality Datasets (VCFs)</h2>

                <table width="80%" style="border-collapse: collapse; font-family: Arial, sans-serif; width: 90%">
                    <tr>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Dataset Name</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Description</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Size</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">*Download time (@100Mbps / @10MBps)</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">MD5 Checksum</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;"><a target="_blank" href="#">chr1_high_quality.vcf.gz</a> </td>
                        <td style="padding: 10px;">High Quality variants and metadata identified in Chromsome 1.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr >
                        <td style="padding: 10px;"><a target="_blank" href="#">chr2_high_quality.vcf.gz</a> </td>
                        <td style="padding: 10px;">High Quality variants and metadata identified in Chromsome 2.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;"><a target="_blank" href="#">chr3_high_quality.vcf.gz</a> </td>
                        <td style="padding: 10px;">High Quality variants and metadata identified in Chromsome 3.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr >
                        <td style="padding: 10px;"><a target="_blank" href="#">chr4_high_Quality.vcf.gz</a> </td>
                        <td style="padding: 10px;">High Quality variants and metadata identified in Chromsome 4.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                </table>
                *Download times are estimated and may vary.

                <h2>F. graminearum 2024 High Quality Datasets (H5)</h2>

                <table width="80%" style="border-collapse: collapse; font-family: Arial, sans-serif; width: 90%">
                    <tr>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Dataset Name</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Description</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Size</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">*Download time (@100Mbps / @10MBps)</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">MD5 Checksum</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;"><a target="_blank" href="#">fusarium2024_chr1_HQ.h5.gz</a> </td>
                        <td style="padding: 10px;">High Quality variants and metadata identified in Chromsome 1.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr >
                        <td style="padding: 10px;"><a target="_blank" href="#">fusarium2024_chr2_HQ.h5.gz</a> </td>
                        <td style="padding: 10px;">High Quality variants and metadata identified in Chromsome 2.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;"><a target="_blank" href="#">fusarium2024_chr3_HQ.h5.gz</a> </td>
                        <td style="padding: 10px;">High Quality variants and metadata identified in Chromsome 3.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr >
                        <td style="padding: 10px;"><a target="_blank" href="#">fusarium2024_chr4_HQ.h5.gz</a> </td>
                        <td style="padding: 10px;">High Quality variants and metadata identified in Chromsome 4.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>

                </table>
                *Download times are estimated and may vary.

                <h2>F. graminearum 2024 Raw Unfiltered Datasets (VCFs)</h2>

                <table width="80%" style="border-collapse: collapse; font-family: Arial, sans-serif; width: 90%">
                    <tr>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Dataset Name</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Description</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Size</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">*Download time (@100Mbps / @10MBps)</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">MD5 Checksum</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;">fusarium2024_chr1.vcf.gz</td>
                        <td style="padding: 10px;">Full unfiltered variants and read data in Chromsome 1.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr >
                        <td style="padding: 10px;">fusarium2024_chr2.vcf.gz</td>
                        <td style="padding: 10px;">Full unfiltered variants and read data in Chromsome 2.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;">fusarium2024_chr3.vcf.gz</td>
                        <td style="padding: 10px;">Full unfiltered variants and read data in Chromsome 3.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>
                    <tr >
                        <td style="padding: 10px;">fusarium2024_chr4.vcf.gz</td>
                        <td style="padding: 10px;">Full unfiltered variants and read data in Chromsome 4.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD</td>
                    </tr>

                </table>
                *Download times are estimated and may vary.

                <h2>Need Help?</h2>
                If you encounter any issues during the download or have any questions, please contact <a href=mailto:john.portwood@usda.gov>John Portwood</a>.

                <h2>Disclaimer</h2>
                <ul>
                    <li>The estimated download times are provided for convenience and can vary based on your network speed and connectivity.</li>
                    <li>We reserve the right to modify the access procedures to ensure data security, integrity, and convenience. This includes using another platform to host the datasets. This page will be updated with the most up-to-date datasets and methods of accessing the data.</li>
                </ul>
            </div>

            <div id="helpContainer" class="content" style="display: none">
                    <h1>Fusarium SNPversity</h1>
                    Release 1.0.01 (28 November 2024)

                    <h2>Citation</h2>
                    If you use SNPversity for your research, please cite one or more of the following
                    <ul class="info-list">
                    <li><b>SNPVerstiy 2.0</b>: <br>Andorf CM, Ross-Ibarra J, Seetharam AS, Hufford MB, Woodhouse MR. (2024) A unified VCF data set from nearly 1,500 diverse maize accessions and resources to explore the genomic landscape of maize. bioRxiv 2024.04.30.591904; <a target="_blank" href="https://doi.org/10.1111/tpj.16123">doi:10.1101/2024.04.30.591904</a>.</li>
                    </ul>

                    <h2>Overview</h2>

                    Welcome to SNPversity 2.0, the second generation of the SNPversity platform. Designed as an open-source visualization tool, SNPversity 2.0 enables users to explore extensive variant datasets with ease. Here's how it works:
                    <ul class="info-list">
                    <li><b>Input & Output</b>: Users can input specific genomic intervals and select the species of interest. SNPversity 2.0 processes this input to deliver a Variant Call Format (VCF) file and a detailed table. These outputs showcase the alleles matching the user's query, providing a clear and concise visualization of genetic variations.</li>
                    <li><b>Technical Architecture</b>: At its core, SNPversity 2.0 is powered by a robust HDF5 database back-end that manages variant annotations efficiently. A data exchange layer, developed in Python, facilitates seamless data handling, while a JavaScript-based interface layer presents Single Nucleotide Polymorphisms (SNPs) and Insertions/Deletions (INDELs) in an interactive format.</li>
                    <li><b>Real-Time Visualization</b>: SNPversity 2.0 elevates data presentation by displaying information in real-time directly in your web browser. Tables are intuitively color-coded, reflecting each SNP's allelic status and mutational state for straightforward interpretation.</li>
                    <li><b>Enhanced Data Insights</b>: Beyond basic variant data, SNPversity 2.0 enriches your research with valuable metadata. This includes variant effect annotations, mapping quality scores, genotypic coverage, and maximum R2 scores to indicate linkage disequilibrium, offering a comprehensive view of genetic variations.</li>
                    <li><b>Genomic Dataset</b>: This version leverages a variant dataset encompassing 513 strains of <i>Fusarium graminearum</i>. These accessions are aligned to the <i>Fusarium graminearum</i> PH-1 (NRRL 31084), ensuring a broad and relevant genetic foundation for analysis.</li>
                    </ul>

                    <h2>Code</h2>
                    For information about the code used in SNPVersity 2.0 please refer to the GitHub page. This page provides detailed information about web compatibility, the directory structure, system requirements, and a comprehensive description of the code. It also offers guidance on how to use the websites, information about the pipelines used, and instructions on preparing data for PanEffect.
                    <br><br>For more details, please visit our GitHub page: <a target="_blank" href="https://github.com/Maize-Genetics-and-Genomics-Database/SNPversity2.0">GitHub - SNPVersity 2.0</a>.

                    <h2><i>F. graminearum</i> 2024 Dataset</h2>

                    TBD
                    <br><br>

                <table width="500px" style="border-collapse: collapse; font-family: Arial, sans-serif;">
                    <tr>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Accessions</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Project</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Project name</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">PubMed ID</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">DOI</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Color Key</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;">513 </td>
                        <td style="padding: 10px;">TBD </td>
                        <td style="padding: 10px;"><i>F. graminearum</i> description.</td>
                        <td style="padding: 10px;">TBD</td>
                        <td style="padding: 10px;">TBD </td>
                        <td style="padding: 10px;"><div style="width: 20px; height: 20px; background-color: #9A5446;"></div></td>
                    </tr>

                    </table>


                <br>
                The table shows the composition of public resequencing data used to build the dataset.  The table lists the number of accessions from each NCBI bioproject. The last column shows the color code used in the table headers in the table view.<br><br>
                The VCF and HDF5 files can be downloaded from the <a target="_blank" href="https://ars-usda.app.box.com/v/maizegdb-public/folder/255392948851">MaizeGDB Box account</a>.
                <br><br>

                    <table width="500px" style="border-collapse: collapse; font-family: Arial, sans-serif;">
                    <tr>
                        <th style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Effect</th>
                        <th style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">High Coverage</th>
                        <th style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">High Quality</th>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;">intergenic</td>
                        <td style="padding: 10px;">1,684,261</td>
                        <td style="padding: 10px;">1,489,039</td>
                    </tr>
                    <tr>
                        <td style="padding: 10px;">5' UTR</td>
                        <td style="padding: 10px;">31,651</td>
                        <td style="padding: 10px;">27,946</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;">synonymous</td>
                        <td style="padding: 10px;">921,773</td>
                        <td style="padding: 10px;">843,174</td>
                    </tr>
                    <tr>
                        <td style="padding: 10px;">missense</td>
                        <td style="padding: 10px;">536,063</td>
                        <td style="padding: 10px;">451,218</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;">stop</td>
                        <td style="padding: 10px;">13,835</td>
                        <td style="padding: 10px;">10,880</td>
                    </tr>
                    <tr>
                        <td style="padding: 10px;">frameshift</td>
                        <td style="padding: 10px;">27,738</td>
                        <td style="padding: 10px;">24,167</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;">intron</td>
                        <td style="padding: 10px;">258,680</td>
                        <td style="padding: 10px;">258,680</td>
                    </tr>
                    <tr>
                        <td style="padding: 10px;">non-coding</td>
                        <td style="padding: 10px;">544</td>
                        <td style="padding: 10px;">367</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px;">3'UTR</td>
                        <td style="padding: 10px;">68,729</td>
                        <td style="padding: 10px;">60,559</td>
                    </tr>
                    <tr>
                        <td style="padding: 10px;">other</td>
                        <td style="padding: 10px;">106</td>
                        <td style="padding: 10px;">83</td>
                    </tr>
                    <tr style="background-color: #F9FBE7;">
                        <td style="padding: 10px; font-weight: bold;">TOTAL</td>
                        <td style="padding: 10px; font-weight: bold;">3,543,380</td>
                        <td style="padding: 10px; font-weight: bold;">3,140,822</td>
                    </tr>
                </table>

                <br>
        <br>
                The table shows the counts for each of the variant effect types for the high-coverage and high-quality datasets.

                    <h2>How to use the website</h2>

                    There are four tabs in the SNPversity 2.0 website 'Select options', 'Table view', 'Tree view', and 'Help'.


                    <h3>Select options</h3>
                    <div style="padding-left: 20px">
                        The main inputs for this tool are selecting the reference genome, genomic interval of interest, and which subset of species to include.

                        <br><br>
                        <b>Select genomic interval:</b><br>
                        This section allows the user to select the genomic interval of interest.  The two main options are entering the genomic coordinates or entering a gene model identifier.

                        <ul>
                            <li><b>Genome Version</b>:  Select the reference genome version.  The only option currently available is B73 version 5. </li>
                            <li><b>Dataset</b>: Select the variation dataset to query.  The '<i>F. graminearum 2024</i> High Quality' dataset was built on 513 different strains of <i>Fusarium graminearum</i> and was filtered on mapping quality (>= 40), genotypic completeness (>= 50%), and linkage disequilibrium with max R2 > 0.5.  There were approximately 3.1 million annotated loci in this dataset.  The '<i>F. graminearum 2024</i> 2024 High Coverage' dataset was built on 513 different strains of <i>Fusarium graminearum</i> and was filtered only on mapping quality (>= 40) and genotypic completeness (>= 50%).  There are approximately 3.5 million annotated loci in this dataset. </li>
                            <li><b>Chromosome</b>: Select the chromosome.  </li>
                            <li><b>Genome Start Position (bp)</b>: Select the start position on the chromosome.</li>
                            <li><b>Genome End Position (bp)</b>: Select the end position on the chromosome.</li>
                            <li><b>Loci per page</b>:  Select the number of loci to view per page in the HTML table view.  This number corresponds to the number of rows in the table. </li>
                            <li><b>Gene Model ID</b>: Optionally, select the gene model identifier and the number of base pairs to add as padding to the start and end of the gene model coordinates.  Pressing the 'Load coordinates' button will load the chromosome, start, and end positions if the gene model is found. This option currently only accepts B73 RefGen_v5 gene models. </li>
                            </ul>
                    (<b>NOTE: Genomic regions larger than 1 MB will only be avaialble as VCF downloads.  The table and tree views will not be available.</b>)
                    <br><br>

                        <b>Select which accessions to include:</b><br>
                        This section allows the user to select a subset of the Fusarium graminearum strains to view. The two main options are to upload a file with the strain IDs, use the buttons to randomly subsample the datasets, or use the checkboxes to manually select the <i>Fusarium</i> strains.  A list of all <i>F. graminearum</i> strains can be found in <a href="fg_accessions.xlsx" download="fg_accessions.xlsx">XLSX</a>.

                    </div>

                    <h3>Table view</h3>
                    <div style="padding-left: 20px">

                        The table view option allows the user to download the VCF generated from the select options tab and displays a table of the data (for regions <= 1Mb). Each row of the data corresponds to a locus position in the dataset. The descriptions of the columns are in the following table.
                        <br><br>

                        <table width="100%" style="border-collapse: collapse; font-family: Arial, sans-serif;">
                            <tr>
                                <th style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Column name</th>
                                <th style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Definition</th>
                                <th style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Abbreviation</th>
                                <th style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Example data</th>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">Chromosome</td>
                                <td style="padding: 10px;">CHR</td>
                                <td style="padding: 10px;">The chromosome where the locus is located</td>
                                <td style="padding: 10px;">chr1</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">Position</td>
                                <td style="padding: 10px;">POS</td>
                                <td style="padding: 10px;">The genomic coordinate on the chromosome</td>
                                <td style="padding: 10px;">104985</td>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">Reference allele</td>
                                <td style="padding: 10px;">REF</td>
                                <td style="padding: 10px;">The allele value for the locus in the reference genome B73.</td>
                                <td style="padding: 10px;">A</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">Alternate allele</td>
                                <td style="padding: 10px;">ALT</td>
                                <td style="padding: 10px;">The alternative allele value found in other species.</td>
                                <td style="padding: 10px;">T</td>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">Gene models</td>
                                <td style="padding: 10px;">Gene model(s)</td>
                                <td style="padding: 10px;">The name of the B73 RefGen_v5 gene model affected by the variant. Displays the closest gene models when the variant is intergenic.</td>
                                <td style="padding: 10px;">Zm00001eb404830</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">Effect type</td>
                                <td style="padding: 10px;">Effect type</td>
                                <td style="padding: 10px;">The type of effect using Sequence Ontology terms.</td>
                                <td style="padding: 10px;">stop gained</td>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">Effect impact</td>
                                <td style="padding: 10px;">Effect impact</td>
                                <td style="padding: 10px;">A estimation of putative impact/deleteriousness.</td>
                                <td style="padding: 10px;">HIGH MODIFIER</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">Mapping quality score</td>
                                <td style="padding: 10px;">MQ</td>
                                <td style="padding: 10px;">The average mapping quality of reads supporting the variant.</td>
                                <td style="padding: 10px;">58</td>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">Completeness</td>
                                <td style="padding: 10px;">COMP</td>
                                <td style="padding: 10px;">The percent of accessions that provide genotype data for a particular variant (i.e., there is at least one read for that accession at the given variant). Note, this is different than read coverage</td>
                                <td style="padding: 10px;">99</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">Maximum squared correlation</td>
                                <td style="padding: 10px;">max R2</td>
                                <td style="padding: 10px;">The linkage disequilibrium measured by the maximum R2 for a given loci.</td>
                                <td style="padding: 10px;">0.64</td>
                            </tr>
                        </table>

                        <br>
                        The gene models in the Gene model(s) column are linked to a genome browser when available.  The position of the locus is shown as a vertical line on the browser.
                        <br><br>
                        Variant effects were calculated using <a target="_blank" href="https://pcingola.github.io/SnpEff/">SNPeff</a>.
                        <br><br>
                        Linkage disequilibrium values were calculated using <a target="_blank" href="https://www.cog-genomics.org/plink/">Plink</a>.
                        <br><br>
                        For synonymous and missense variant effect types, the information in the Effect type column shows the following information in parentheses: the amino acid for the reference genome, the position, and the amino acid substitution in the alternative genome (e.g. G477S). Missense variant effect types are linked to the <a target="_blank" href="https://www.maizegdb.org/effect/fusarium/index.php">Fusarium Protein Toolkit (FPT)</a>.
                        <br><br>
                        The remainder of the columns are based on the subset of selected <i>Fusarium</i> species.  There is one column for each <i>Fusarium</i> species and the column header is color-coded based on the project.  The columns are named based on the accession name, an underscore, and the SRR ID.  The values and colors of the data in these columns are based on the allele value for the given locus for that accession.

                        <ul style="list-style-type: none; padding: 0;">
                            <li style="margin-bottom: 5px; font-size: 14px; line-height: 20px;">
                                <span class="color-box" style="display: inline-block; width: 20px; height: 20px; margin-right: 10px; vertical-align: middle; background-color: #8CC28D; text-align: center; line-height: 20px;">0</span>
                                Reference genotype.
                            </li>

                            <li style="margin-bottom: 5px; font-size: 14px; line-height: 20px;">
                                <span class="color-box" style="display: inline-block; width: 20px; height: 20px; margin-right: 10px; vertical-align: middle; background-color: #FD950A; text-align: center; line-height: 20px;">1</span>
                                Alternate genotype.
                            </li>

                            <li style="margin-bottom: 5px; font-size: 14px; line-height: 20px;">
                                <span class="color-box" style="display: inline-block; width: 20px; height: 20px; margin-right: 10px; vertical-align: middle; background-color: #AAAAAA; text-align: center; line-height: 20px;">N</span>
                                Missing or unknown genotypes
                            </li>
                        </ul>

                    </div>

                    <h3>Variant effect types</h3>
                    <div style="padding-left: 20px">

                            <table width="1200px" style="border-collapse: collapse; font-family: Arial, sans-serif;">
                                <tr>
                                    <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Seq. Ontology</td>
                                    <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Effect</td>
                                    <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Description</td>
                                    <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Impact</td>
                                </tr>
                                <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">intergenic_region</td>
                                <td style="padding: 10px;">INTERGENIC</td>
                                <td style="padding: 10px;">The variant is in an intergenic region</td>
                                <td style="padding: 10px;">MODIFIER</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">upstream_gene_variant</td>
                                <td style="padding: 10px;">UPSTREAM</td>
                                <td style="padding: 10px;">Upstream of a gene (default length: 5K bases)</td>
                                <td style="padding: 10px;">MODIFIER</td>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">5_prime_UTR_variant</td>
                                <td style="padding: 10px;">UTR_5_PRIME</td>
                                <td style="padding: 10px;">Variant hits 5'UTR region</td>
                                <td style="padding: 10px;">MODIFIER</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">coding_sequence_variant</td>
                                <td style="padding: 10px;">CDS</td>
                                <td style="padding: 10px;">The variant hits a CDS.</td>
                                <td style="padding: 10px;">MODIFIER</td>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">exon_variant</td>
                                <td style="padding: 10px;">EXON</td>
                                <td style="padding: 10px;">The variant hits an exon (from a non-coding transcript) or a retained intron.</td>
                                <td style="padding: 10px;">MODIFIER</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">intron_variant</td>
                                <td style="padding: 10px;">INTRON</td>
                                <td style="padding: 10px;">Variant hits an intron. Technically, hits no exon in the transcript.</td>
                                <td style="padding: 10px;">MODIFIER</td>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">frameshift_variant</td>
                                <td style="padding: 10px;">FRAME_SHIFT</td>
                                <td style="padding: 10px;">Insertion or deletion causes a frame shift e.g.: An indel size is not multiple of 3</td>
                                <td style="padding: 10px;">HIGH</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">missense_variant</td>
                                <td style="padding: 10px;">NON_SYNONYMOUS_CODING</td>
                                <td style="padding: 10px;">Variant causes a codon that produces a different amino acid e.g.: Tgg/Cgg, W/R</td>
                                <td style="padding: 10px;">MODERATE</td>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">synonymous_variant</td>
                                <td style="padding: 10px;">SYNONYMOUS_CODING</td>
                                <td style="padding: 10px;">Variant causes a codon that produces the same amino acid e.g.: Ttg/Ctg, L/L</td>
                                <td style="padding: 10px;">LOW</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">stop_lost</td>
                                <td style="padding: 10px;">STOP_LOST</td>
                                <td style="padding: 10px;">Variant causes stop codon to be mutated into a non-stop codon e.g.: Tga/Cga, */R</td>
                                <td style="padding: 10px;">HIGH</td>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">stop_gained</td>
                                <td style="padding: 10px;">STOP_GAINED</td>
                                <td style="padding: 10px;">Variant causes a STOP codon e.g.: Cag/Tag, Q/*</td>
                                <td style="padding: 10px;">HIGH</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">3_prime_UTR_variant</td>
                                <td style="padding: 10px;">UTR_3_PRIME</td>
                                <td style="padding: 10px;">Variant hits 3'UTR region</td>
                                <td style="padding: 10px;">MODIFIER</td>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">downstream_gene_variant</td>
                                <td style="padding: 10px;">DOWNSTREAM</td>
                                <td style="padding: 10px;">Downstream of a gene (default length: 5K bases)</td>
                                <td style="padding: 10px;">MODIFIER</td>
                            </tr>
                        </table>
                        <br><br>

                        <table width="1200px" style="border-collapse: collapse; font-family: Arial, sans-serif;">
                            <tr>
                                <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Impact</td>
                                <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Meaning</td>
                                <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Example</td>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">HIGH</td>
                                <td style="padding: 10px;">The variant is assumed to have high (disruptive) impact in the protein, probably causing protein truncation, loss of function or triggering nonsense mediated decay.</td>
                                <td style="padding: 10px;">stop_gained, frameshift_variant</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">MODERATE</td>
                                <td style="padding: 10px;">A non-disruptive variant that might change protein effectiveness.</td>
                                <td style="padding: 10px;">missense_variant, inframe_deletion</td>
                            </tr>
                            <tr style="background-color: #F9FBE7;">
                                <td style="padding: 10px;">LOW</td>
                                <td style="padding: 10px;">Assumed to be mostly harmless or unlikely to change protein behavior.</td>
                                <td style="padding: 10px;">synonymous_variant</td>
                            </tr>
                            <tr>
                                <td style="padding: 10px;">MODIFIER</td>
                                <td style="padding: 10px;">Usually non-coding variants or variants affecting non-coding genes, where predictions are difficult or there is no evidence of impact.</td>
                                <td style="padding: 10px;">exon_variant, downstream_gene_variant</td>
                            </tr>
                        </table>


                        <br>

                        To see a full list of the variant effect types click <a target="_blank" href="https://pcingola.github.io/SnpEff/snpeff/inputoutput/">here</a>.
                    </div>

                    <h3>VCF Download from Search</h3>
                    <div style="padding-left: 20px">
                        After completing a search based on a genomic region and a subset of accessions, a "Download the VCF file" button will be available in the "Table view" tab. When the user clicks this link, a subset of the variants meeting the search criteria will be downloaded as a VCF file. The VCF will only include the specified set of <i>Fusarium</i> species and each variant within the selected genomic region. The VCF will contain rows with no variant relative to the reference. The genotype coverage fields (CVC and CVP) will refer to the number of accessions with genotype data for a particular variant based on all 1,525 accessions, not the local genotype coverage of the selected accessions.
                        <br><br>
                        Below is a sample of an output VCF file generated from SNPVersity 2.0:

<pre>
##fileformat=VCFv4.2
##fileDate=20241129
##source=Fusarium2024
##reference=TBD
##INFO=&lt;ID=MQ,Number=1,Type=Float,Description="RMS mapping quality."&gt;
##INFO=&lt;ID=CVC,Number=1,Type=Integer,Description="The number of accessions that have genotype data for a particular variant."&gt;
##INFO=&lt;ID=CVP,Number=1,Type=Float,Description="The percent of accessions that have genotype data for a particular variant."&gt;
##INFO=&lt;ID=TYPE,Number=.,Type=String,Description="The type of effect using Sequence Ontology terms."&gt;
##INFO=&lt;ID=EFFECT,Number=.,Type=String,Description="An estimation of putative impact/deleteriousness."&gt;
##INFO=&lt;ID=GENEMODEL,Number=.,Type=String,Description="The name of the gene model affected by the variant."&gt;
##INFO=&lt;ID=SUB,Number=.,Type=String,Description="The amino acid substitution for missense and non-synonymous variants."&gt;
##INFO=&lt;ID=MAXR2,Number=1,Type=Float,Description="The maximum R2 for a given loci."&gt;
##FORMAT=<&lt;ID=GT,Number=1,Type=String,Description="Genotype"&gt;
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	45596mAP	45668mAP	45702mAP	46290mAP	46370mAP	46566mAP	46960mAP	ETH-001-barcode9mAP	F100mAP	F272mAP	F325mAP	F328mAP	FHK15mAP	FHK28mAP	FHK45mAP	FHK50mAP	FHK69mAP	MTD69mAP	SAMN16774488mAP	SAMN16774508mAP	SAMN16774523mAP	SAMN16774539mAP	SAMN16774540mAP	SAMN16774547mAP	WG15Evill16_S33mAP	WG15GG39_S5mAP
chr4	555005	.	C	T	3609.51	.	MQ=60;CVC=512;CVP=99.81;TYPE=synonymous_variant,downstream_gene_variant;EFFECT=LOW,MODIFIER;GENEMODEL=FGSG_06605,FGSG_12945;SUB=L886L,.;MAXR2=1.0	GT	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0
chr4	555008	.	G	A	856.84	.	MQ=60;CVC=512;CVP=99.81;TYPE=synonymous_variant,downstream_gene_variant;EFFECT=LOW,MODIFIER;GENEMODEL=FGSG_06605,FGSG_12945;SUB=D885D,.;MAXR2=1.0	GT	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0
chr4	555011	.	G	T	3872.84	.	MQ=60;CVC=512;CVP=99.81;TYPE=synonymous_variant,downstream_gene_variant;EFFECT=LOW,MODIFIER;GENEMODEL=FGSG_06605,FGSG_12945;SUB=A884A,.;MAXR2=1.0	GT	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0
chr4	555041	.	G	A	4066.84	.	MQ=60;CVC=512;CVP=99.81;TYPE=synonymous_variant,downstream_gene_variant;EFFECT=LOW,MODIFIER;GENEMODEL=FGSG_06605,FGSG_12945;SUB=G874G,.;MAXR2=1.0	GT	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0
chr4	555089	.	G	A	10879.3	.	MQ=60;CVC=512;CVP=99.81;TYPE=synonymous_variant,downstream_gene_variant;EFFECT=LOW,MODIFIER;GENEMODEL=FGSG_06605,FGSG_12945;SUB=N858N,.;MAXR2=0.798422	GT	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0
</pre>

                    </div>

                    <h3>Tree view</h3>
                    <div style="padding-left: 20px">
                        The tree view option allows the user to generate phylogenetic tree views based on the <a target="_blank" href="https://github.com/sansubs/vcf2pop">VCF2PopTree tool</a>.  The input to build the tree is based on the VCF file generated from the select options tab.
                        <br><br>
                        The tree tree can be constructed as either an UPGMA tree or Neighbour-Joining tree (Unrooted).  The drawing options are inlcudes Rectangular tree or Radial tree.  In addtion, the trees can be saved in the following text formats: Newick tree, Pair-wise diversity (MEGA), or PHYLIP.

                    </div>

                    <h3>Downloads</h3>
                    <div style="padding-left: 20px">
                        The Downloads tab provides detailed information about the datasets available for download, along with guidelines and best practices.
                    </div>

                    <h3>Help</h3>
                    <div style="padding-left: 20px">
                    The help page gives an overview of the webpage, descriptions of the datasets and methods, and how to use the website.
                    </hgroup>
                    </div>

                </div>
                </div>

            <div id="tableToPopupM" style="display:none;">
            <table width="800px" style="border-collapse: collapse; font-family: Arial, sans-serif; font-size: 12px;">
                <tr>
                    <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Accessions</td>
                    <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Project</td>
                    <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Project name</td>
                    <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Color Key</td>
                </tr>
                <tr>
                    <td style="padding: 10px;">513</td>
                    <td style="padding: 10px;"><a target="_blank" href="https://www.ncbi.nlm.nih.gov/bioproject/???">???</a> </td>
                    <td style="padding: 10px;">F. graminearum description.</td>
                    <td style="padding: 10px;"><div style="width: 20px; height: 20px; background-color: #9A5446;"></div></td>
                </tr>

                </table>
    </div>


                <div id="tableToPopupS" style="display:none;">
                <table width="800px" style="border-collapse: collapse; font-family: Arial, sans-serif; font-size: 12px;">
                    <tr>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Accessions</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Project</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Project name</td>
                        <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Color Key</td>
                    </tr>

                    </table>
        </div>

    <div id="alleleToPopup" style="display:none;">
        <ul style="list-style-type: none; padding: 0;">
            <li style="margin-bottom: 5px; font-size: 20px; line-height: 40px;">
                <span class="color-box" style="display: inline-block; width: 40px; height: 40px; margin-right: 10px; vertical-align: middle; background-color: #8CC28D; text-align: center; line-height: 40px;">0</span>
                Reference genotype.
            </li>

            <li style="margin-bottom: 5px; font-size: 20px; line-height: 40px;">
                <span class="color-box" style="display: inline-block; width: 40px; height: 40px; margin-right: 10px; vertical-align: middle; background-color: #FD950A; text-align: center; line-height: 40px;">1</span>
                Alternate genotype.
            </li>

            <li style="margin-bottom: 5px; font-size: 20px; line-height: 40px;">
                <span class="color-box" style="display: inline-block; width: 40px; height: 40px; margin-right: 10px; vertical-align: middle; background-color: #AAAAAA; text-align: center; line-height: 40px;">N</span>
                Missing or unknown genotypes
            </li>
        </ul>
    </div>

    <div id="varToPopup" style="display:none;">
    <table width="800px" style="border-collapse: collapse; font-family: Arial, sans-serif; font-size: 12px;">
        <tr>
            <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Seq. Ontology</td>
            <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Effect</td>
            <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Description</td>
            <td align="center" style="border-bottom: 3px solid #4CAF50; padding: 10px; background-color: #E8F5E9; font-weight: bold;">Impact</td>
        </tr>
        <tr style="background-color: #F9FBE7;">
        <td style="padding: 10px;">intergenic_region</td>
        <td style="padding: 10px;">INTERGENIC</td>
        <td style="padding: 10px;">The variant is in an intergenic region</td>
        <td style="padding: 10px;">MODIFIER</td>
    </tr>
    <tr>
        <td style="padding: 10px;">upstream_gene_variant</td>
        <td style="padding: 10px;">UPSTREAM</td>
        <td style="padding: 10px;">Upstream of a gene (default length: 5K bases)</td>
        <td style="padding: 10px;">MODIFIER</td>
    </tr>
    <tr style="background-color: #F9FBE7;">
        <td style="padding: 10px;">5_prime_UTR_variant</td>
        <td style="padding: 10px;">UTR_5_PRIME</td>
        <td style="padding: 10px;">Variant hits 5'UTR region</td>
        <td style="padding: 10px;">MODIFIER</td>
    </tr>
    <tr>
        <td style="padding: 10px;">coding_sequence_variant</td>
        <td style="padding: 10px;">CDS</td>
        <td style="padding: 10px;">The variant hits a CDS.</td>
        <td style="padding: 10px;">MODIFIER</td>
    </tr>
    <tr style="background-color: #F9FBE7;">
        <td style="padding: 10px;">exon_variant</td>
        <td style="padding: 10px;">EXON</td>
        <td style="padding: 10px;">The variant hits an exon (from a non-coding transcript) or a retained intron.</td>
        <td style="padding: 10px;">MODIFIER</td>
    </tr>
    <tr>
        <td style="padding: 10px;">intron_variant</td>
        <td style="padding: 10px;">INTRON</td>
        <td style="padding: 10px;">Variant hits an intron. Technically, hits no exon in the transcript.</td>
        <td style="padding: 10px;">MODIFIER</td>
    </tr>
    <tr style="background-color: #F9FBE7;">
        <td style="padding: 10px;">frameshift_variant</td>
        <td style="padding: 10px;">FRAME_SHIFT</td>
        <td style="padding: 10px;">Insertion or deletion causes a frame shift e.g.: An indel size is not multiple of 3</td>
        <td style="padding: 10px;">HIGH</td>
    </tr>
    <tr>
        <td style="padding: 10px;">missense_variant</td>
        <td style="padding: 10px;">NON_SYNONYMOUS_CODING</td>
        <td style="padding: 10px;">Variant causes a codon that produces a different amino acid e.g.: Tgg/Cgg, W/R</td>
        <td style="padding: 10px;">MODERATE</td>
    </tr>
    <tr style="background-color: #F9FBE7;">
        <td style="padding: 10px;">synonymous_variant</td>
        <td style="padding: 10px;">SYNONYMOUS_CODING</td>
        <td style="padding: 10px;">Variant causes a codon that produces the same amino acid e.g.: Ttg/Ctg, L/L</td>
        <td style="padding: 10px;">LOW</td>
    </tr>
    <tr>
        <td style="padding: 10px;">stop_lost</td>
        <td style="padding: 10px;">STOP_LOST</td>
        <td style="padding: 10px;">Variant causes stop codon to be mutated into a non-stop codon e.g.: Tga/Cga, */R</td>
        <td style="padding: 10px;">HIGH</td>
    </tr>
    <tr style="background-color: #F9FBE7;">
        <td style="padding: 10px;">stop_gained</td>
        <td style="padding: 10px;">STOP_GAINED</td>
        <td style="padding: 10px;">Variant causes a STOP codon e.g.: Cag/Tag, Q/*</td>
        <td style="padding: 10px;">HIGH</td>
    </tr>
    <tr>
        <td style="padding: 10px;">3_prime_UTR_variant</td>
        <td style="padding: 10px;">UTR_3_PRIME</td>
        <td style="padding: 10px;">Variant hits 3'UTR region</td>
        <td style="padding: 10px;">MODIFIER</td>
    </tr>
    <tr style="background-color: #F9FBE7;">
        <td style="padding: 10px;">downstream_gene_variant</td>
        <td style="padding: 10px;">DOWNSTREAM</td>
        <td style="padding: 10px;">Downstream of a gene (default length: 5K bases)</td>
        <td style="padding: 10px;">MODIFIER</td>
    </tr>
</table>
<br><br>
<table style="border-collapse: collapse; font-family: Arial, sans-serif; font-size: 12px;">
    <tr>
        <th>Impact</th>
        <th>Meaning</th>
        <th>Example</th>
    </tr>
    <tr>
        <td style="padding: 10px;">HIGH</td>
        <td style="padding: 10px;">The variant is assumed to have high (disruptive) impact in the protein, probably causing protein truncation, loss of function or triggering nonsense mediated decay.</td>
        <td style="padding: 10px;">stop_gained, frameshift_variant</td>
    </tr>
    <tr>
        <td style="padding: 10px;">MODERATE</td>
        <td style="padding: 10px;">A non-disruptive variant that might change protein effectiveness.</td>
        <td style="padding: 10px;">missense_variant, inframe_deletion</td>
    </tr>
    <tr>
        <td style="padding: 10px;">LOW</td>
        <td style="padding: 10px;">Assumed to be mostly harmless or unlikely to change protein behavior.</td>
        <td style="padding: 10px;">synonymous_variant</td>
    </tr>
    <tr>
        <td style="padding: 10px;">MODIFIER</td>
        <td style="padding: 10px;">Usually non-coding variants or variants affecting non-coding genes, where predictions are difficult or there is no evidence of impact.</td>
        <td style="padding: 10px;">exon_variant, downstream_gene_variant</td>
    </tr>
</table>
<br><br>

To see a full list of the variant effect types click <a target="_blank" href="https://pcingola.github.io/SnpEff/snpeff/inputoutput/">here</a>.

    </div>

        <div class="footer">
            Site designed and hosted by <a target="_blank" href="https://www.maizegdb.org/">MaizeGDB</a> with support from the USDA<br>
        </div>
    </body>
    </html>
